### PR TITLE
Hardening/warnings as errors in ubuntu 1704

### PR DESCRIPTION
**Pull request by kzangeli**
Better compilation flags for Ubuntu 17.04.
The -Werror flag was removed for Ubuntu 17.04 as its newer compiler have plenty of **depracated** warnings.

Now,  -Werror has been put back and instead of not using -Werror to fix the problem, a new flag has been added: -Wno-deprecated-declarations.

So, now we detect warnings in Ubuntu 17.04 again (just not deprecated warnings)

A few new warnings have surged in Ubuntu 17.04 (in lmlib) and these warnings are fixed as well in this PR - otherwise it doesn't compile!

### CODE REVIEW

**Comment by fgalan** [comment]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -75,7 +75,7 @@ add_definitions(-fPIC)
 # A grep of 'std::auto_ptr<' in the header files show 7 hits, and 13 hits in cpp files
 #
 IF (${DISTRO} STREQUAL "Ubuntu_17.04")
```
Comment (fgalan) [inline CMakeLists.txt]: The comment just above these lines should be reviewed to check if valid or obsolete.

Comment (kzangeli) [reply CMakeLists.txt]: Fixed in b9173da

**Comment by fgalan** [comment]
LGTM. If your pass for ft is ok, ready to be merged.

**Pull request merged**